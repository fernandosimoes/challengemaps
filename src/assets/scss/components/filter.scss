%defaultinput {
  border-radius: 5px;
  color: $gray;
  font-size: 20px;
  height: 35px;
  margin-bottom: 15px;
  padding: 0 15px;
  text-align: right;
  width: 320px;

  &::placeholder {
    color: $lightgray;
  }
}

.minimumvalue {
  label {
    font-family: $latobold
  }
}

.minimumvalueinput {
  @extend %defaultinput;
}

.searchinput {
  @extend %defaultinput;
  align-self: center;
  text-align: left;
  width: 100%;
}

.filters {
  justify-content: space-between;
  width: 100%;

  .search {
    align-items: center;
    display: flex;
    width: 100%;
  }
}

.searchinput {
  background: no-repeat 98% center;
  margin-top: 28px;
}

.faturamentominimo {
  align-items: center;
  display: flex;
  font-size: 18px;
  justify-content: space-between;

  label {
    margin-bottom: 15px;
  }

  button {
    margin-top: 10px;
  }

  span {
    color: $red;
    font-size: 12px;
    margin: -15px 0 15px;
  }
}