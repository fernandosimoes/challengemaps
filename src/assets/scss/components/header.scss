.header {
  background: $blue;
  font-size: 35px;
  margin-bottom: 25px;
  width: 100%;

  h1 {
    color: $white;
    font-family: $lato;
    padding: 25px;
  }
}