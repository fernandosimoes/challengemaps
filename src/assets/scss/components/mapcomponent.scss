.mapcontent {
    width: 40%;
}

.minimal--value {
    border-radius: 5px;
    color: $gray;
    height: 35px;
    margin-bottom: 15px;
    text-align: right;
    width: 320px;
    padding: 0 15px;
    font-size: 20px;
    &::placeholder {
        color: $lightgray;
    }
}

.faturamentominimo {
    display: flex;
    flex-direction: column;
    font-size: 25px;

    label {
        margin-bottom: 15px;
    }

    span {
        margin: -15px 0 15px;
        font-size: 12px;
        color: $red;
    }
}