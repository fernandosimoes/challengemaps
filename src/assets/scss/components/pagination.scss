.pagination {

  button {
    background: none;
    border: none;
    cursor: pointer;

    &.selected {
      background-color: $lightgray;

    }
  }
}
