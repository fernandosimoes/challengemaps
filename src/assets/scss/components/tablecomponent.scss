.tablecontent {
    width: 100%;
}

.table {
    border-bottom: 0;
    border-radius: 5px;
    border: 1px solid $gray;

    .line {
        border-bottom: 1px solid $gray;
        display: flex;
        justify-content: space-between;
        width: 100%;
        span {
            font-size: 18px;
            margin:0 5px;
            padding: 5px 0;
        }

        a {
            font-size: 18px;
            margin:0 5px;
            padding: 5px 0;
            text-decoration: none;
            color: $gray;
            font-weight: bold;
        }

        .low {
            color: $red
        }
    }
}