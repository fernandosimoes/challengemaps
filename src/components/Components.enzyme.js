// Link.react.test.js
import React from 'react';
import Header from './Header';
import { MapContainer } from './Map';

import { configure, shallow, mount, render } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('Enzyme Map', () => {
    // Render a checkbox with label in the document
    const map = shallow(<MapContainer />);
    // expect(header.hasClass('header')).toBe(true);
    // expect(wrapper.find('.my-button').hasClass('disabled')).to.equal(true);
    // expect(checkbox.text()).toEqual('Off');

    // checkbox.find('input').simulate('change');

    // expect(checkbox.text()).toEqual('On');
});