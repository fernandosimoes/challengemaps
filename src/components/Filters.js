import React, { Component } from 'react';

class Filters extends Component {
    render() {
        return (
            <div className="Filters">
                <div className="faturamentominimo">
                    <label htmlFor="">Faturamento minimo esperado</label>
                    <input type="text" className="minimal--value" placeholder='' />
                    {/* <span className="warning" className={errorclass}>
                        just number are acepted.
                    </span> */}
                </div>
                <div className="faturamentominimo">
                    <label htmlFor="">Faturamento minimo esperado</label>
                    <input type="text" className="minimal--value" placeholder='' />
                    {/* <span className="warning" className={errorclass}>
                        just number are acepted.
                    </span> */}
                </div>
            </div>
        );
    }
}

export default Filters;