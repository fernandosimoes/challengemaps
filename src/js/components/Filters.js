import React, { Component } from 'react';
import { connect } from 'react-redux';
import { changeminimumvalue, filterstores, showPageStoreonMap } from "../actions/lojasaction";
import CurrencyFormat from "react-currency-format";
import searchimage from  "../../assets/images/search.svg";

const stylebgsearch = {
    backgroundImage: `url(${searchimage})`,
}


class Filters extends Component {
    constructor(props) {
        super(props);
        this.filterprice = this.filterprice.bind(this)
        this.filterbyname = this.filterbyname.bind(this)
        this.filterpagesonmap = this.filterpagesonmap.bind(this)
    }

    filterprice(e) {
        this.props.changeminimumvalue(e.target.value)
    }

    filterbyname (e) {


        this.props.filterstores(String(e.target.value).toLocaleLowerCase(), this.props.todaslojas);
    }

    filterpagesonmap () {
        const newvalue = this.props.pagesonmap ? false : true;
        this.props.showPageStoreonMap(newvalue);
    }

    render() {
        const placeholder = () => (<CurrencyFormat  value={this.props.minimumvalue} Type={'text'} thousandSeparator={true} prefix={'R$ '} />);
        return (
            <div className="filter">
                <div className="columns inputs">
                    <div className="column search">
                        <input type="text" className="searchinput" style={stylebgsearch} value={this.props.filtro} placeholder="Pesquisa" onChange={this.filterbyname}  />
                        {/* <span className="warning" className={errorclass}>
                            just number are acepted.
                        </span> */}
                    </div>
                    <div className="column faturamentominimo">
                        <div className="minimumvalue">
                            <label htmlFor="">Faturamento minimo esperado</label><br />
                            <input type="text" className="minimumvalueinput" onChange={this.filterprice} placeholder={this.props.minimumvalue}/>
                        </div>
                        {!this.props.pagesonmap && <button className="button" onClick={this.filterpagesonmap}>Mostrar apenas lojas da tabela</button>}
                        {this.props.pagesonmap && <button className="button" onClick={this.filterpagesonmap}>Mostrar todas lojas no mapa</button>}
                    </div>
                </div>

            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        lojas: state.lojas,
        minimumvalue: state.minimumvalue,
        todaslojas: state.todaslojas,
        filtro: state.filtro,
        pagesonmap: state.pagesonmap
    }
}


const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        changeminimumvalue: (value) => { dispatch(changeminimumvalue(value))},
        filterstores: (value, stores) => { dispatch(filterstores(value, stores))},
        showPageStoreonMap: (value) => { dispatch(showPageStoreonMap(value))}
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(Filters)
// export default Filters;
