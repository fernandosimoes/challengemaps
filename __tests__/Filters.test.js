// Link.react.test.js
import React from 'react';
import { FiltersComponent } from '../src/js/components/Filters';
import renderer from 'react-test-renderer';
import {allstores} from './fakedata'

test('Filters', () => {

    let component = renderer.create(
        <FiltersComponent pagesonmap={false} lojas={allstores.stores} />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
