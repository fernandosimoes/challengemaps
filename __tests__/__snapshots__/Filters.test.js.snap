// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters 1`] = `
<div
  className="filter"
>
  <div
    className="columns inputs"
  >
    <div
      className="column search"
    >
      <input
        className="searchinput"
        onChange={[Function]}
        placeholder="Pesquisa"
        style={
          Object {
            "backgroundImage": "url()",
          }
        }
        type="text"
      />
    </div>
    <div
      className="column faturamentominimo"
    >
      <div
        className="minimumvalue"
      >
        <label
          htmlFor=""
        >
          Faturamento minimo esperado
        </label>
        <br />
        <input
          className="minimumvalueinput"
          onChange={[Function]}
          type="text"
        />
      </div>
      <button
        className="button"
        onClick={[Function]}
      >
        Mostrar apenas lojas da tabela
      </button>
    </div>
  </div>
</div>
`;
